src/client/components/controls/clickable-edit.js

```diff
@@ -5,70 +5,89 @@ import {
 import PropTypes from 'prop-types';
```

Comment:
This component is still more complex than I'd prefer, even with the refactoring. Since the project could well have a wide variety of other editing scenarios, it may be worthwhile to try to find even cleaner patterns for expressing the concept of async editing.